### PR TITLE
Testing and validation

### PR DESCRIPTION
**Pull request by kennyegun24**
### In this pr, I did the following...
- I wrote unit tests for the methods
- I validated user inputs based on projects requirements

### CODE REVIEW

**Comment by mirouhml** [comment]
Hi @kennyegun24, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that didn't pass linters checks.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/model/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  it 'user bio to match' do
+    @user.Bio = 'Kenny is a boy'
+    expect(@user).to be_valid
+  end
+
+  it 'Check if postscounter is an integer' do
+    @user.PostCounter = 'Random'
+    expect(@user).to_not be_valid
+  end
+end
```
Comment (mirouhml) [inline spec/model/users_spec.rb]: - [x] Kindly add a unit test for the model method `three_posts`, as it is required to do so.

File: spec/model/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  it 'Check if CommentCounter is an integer' do
+    @post.CommentCounter = 'Random'
+    expect(@post).to_not be_valid
+  end
+
+  it 'Check if like count is an integer' do
+    @post.LikesCounter = 'data'
+    expect(@post).to_not be_valid
+  end
+end
```
Comment (mirouhml) [inline spec/model/posts_spec.rb]: - [x] Same as before, kindly add unit tests for post model methods, as it is required to do so.

**Review by mirouhml — CHANGES_REQUESTED** [review]
# REQUIRED CHANGES 🔴

Hi @kennyegun24, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  before :each do
+    @user = User.new(id: 1, Name: 'Kenny', Bio: 'I am the greatest being alive', PostCounter: 1,
+                     Photo: 'https://google.com/googleIcon.png')
```
Comment (BenMukebo) [inline spec/model/comment_spec.rb:6]: - [OPTIONAL] Well done writing your validations tests so far. However, you're missing a small thing to do for the tests to be valid. Kindly make all Model specs (for `User`, `Post`, `Comment`, and `Like` model) has **Validation** tests as required in this milestone; See the example down below!
```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

File: spec/model/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @user = User.new(id: 1, Name: 'Kenny', Bio: 'I am the greatest being alive', PostCounter: 1,
+                     Photo: 'https://google.com/googleIcon.png')
+
+    @post = Post.new(title: 'Love', text: 'My everthing', author_id: 1, CommentCounter: 1, LikesCounter: 0)
+  end
+
+  it 'title must not be blank' do
+    @post.title = nil
+    expect(@post).to_not be_valid
+  end
```
Comment (BenMukebo) [inline spec/model/posts_spec.rb:14]: - [ ] According to the project requirement, 👇🏻 Any custom methods in the models should have a corresponding unit test. You're missing adding unit test of the:
1.  `update_post_counter and five_recent_comments` methods of your post model.
2. `update_comment_counter` method of your comment model and,
3. `update_likes_counter` method of your like model.
### please address this ✍🏻 

im2

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @kennyegun24,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Descriptive Pull Request.  ✔️

 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] When I click the link to the `MIT.md` file that you added, I get this 404 `error`:

00Mit

- Please, update your LICENSE section with a link to the license file, as this will make your readme more informative and professional.


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi 🙋  @kennyegun24, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏 

- Add tests for custom methods 👍 
- PR is professional 👍 

## Optional suggestions

- **@kennyegun24, kindly note that broken links hurt user experience as requested by the previous reviewer and it makes your work looks unprofessional implying that you don't care about those using or checking your work. The license link is still not working**. It is highly recommended to fix it!

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**